bin/zeppelin-daemon.sh

```diff
@@ -98,6 +98,21 @@ function initialize_default_directories() {
     echo "Pid dir doesn't exist, create ${ZEPPELIN_PID_DIR}"
     $(mkdir -p "${ZEPPELIN_PID_DIR}")
   fi
+
+  # TODO(jl): We need to discuss if we keep this feature or not for the future. See ZEPPELIN-3366
+  if [[ -z "${SPARK_HOME}" ]] && [[ "${SPARK_DISABLE_DOWNLOAD}" != "true" ]]; then
+    # I didn't make it configurable because it's not decided to be open to public
+    SPARK_DIST_DIR="${ZEPPELIN_HOME}/.spark-dist"
+    if [[ ! -d ${SPARK_DIST_DIR} ]]; then
+      mkdir -p ${SPARK_DIST_DIR}
+      URL=$(curl -fsSL https://www.apache.org/dyn/closer.lua/spark/spark-2.3.0/spark-2.3.0-bin-hadoop2.7.tgz | grep -o '<strong>[^<]*</strong>' | sed 's/<[^>]*>//g' | head -1)
```

Comment:
how do we handle different version of spark?

Reply:
My intention was it's only for those who don't want to download Spark by themselves. We could make it configurable as same as the way to set `SPARK_VER` and `HADOOP_VER` like `testing/downlaodSpark.sh`. Do you think it's possible to be configurable?

Reply:
hmm, also if I have SPARK_HOME I don't need to have .spark-dist, right?

Reply:
Sure, To keep current behavior, this PR will download spark distribution. If you set one of `SPARK_HOME` and `SPARK_DISABLE_DOWNLOAD`, the download step will be ignored.

Reply:
In case of downloading Spark distribution, `.spark-dist` will be used as a default directory to store it.

Reply:
ok got it

Reply:
I think the problem isn't multiple version, but with a hardcoded version + favor - it's easy for us to forget about this when we move to spark 2.4? or 2.3.1?

Reply:
Yes, right. I'm also a bit afraid of it. This kind of hardcoded script doesn't look better. Let me find another way of handling it.

Reply:
Downloading spark in zeppelin-daemons.h would make zeppelin server start pretty slow I think, and users may think zeppelin fails to start.

Reply:
Actually, `bin/zeppelin.sh` and `bin/zeppelin-daemon.sh` will show the message Spark is downloading. In case of advanced users and those who already have their own Spark package, they can set `SPARK_DISABLE_DOWNLOAD` into zeppelin-env.sh, which helps to avoid downloading Spark distribution when starting it. BTW, do you have any idea of downloading Spark distribution?

Reply:
How about downloading spark when user start spark interpreter at the first time ? AFAIK, some users don't use spark in zeppelin, they may only use jdbc or other interpreters.